src/sidenav/sidenav-item.tsx

```diff
@@ -28,17 +46,98 @@ export const SidenavItem = ({
         setSelectedItem,
     } = useContext(SidenavContext);
 
-    // =============================================================================
+    const id = otherProps.id || title.toLowerCase().replaceAll(" ", "-");
+    const drawerId = `${internalId}-drawer`;
+    const isSelected = !!selectedItem && selectedItem.itemId === id;
+    const isCurrent = !!currentItem && currentItem.itemId === id;
+    const [isOpen, setIsOpen] = useState(false);
+
+    const [anchorRef, setAnchorRef] = useState<HTMLButtonElement | null>(null);
+    const [drawerRef, setDrawerRef] = useState<HTMLDivElement | null>(null);
+
+    // =========================================================================
+    // FLOATING UI
+    // =========================================================================
+    const { floatingStyles, refs, context } = useFloating({
+        open: isOpen,
+        onOpenChange: (open: boolean, _event, reason: OpenChangeReason) => {
+            if (open) {
+                if (menuRef.current) {
+                    refs.setPositionReference(menuRef.current);
```

Comment:
floating ui tracks three elements:
- floating (the popup that typically overlays page content)
- reference (which is the trigger for interactions)
- position reference (the anchor that the floating element is positioned relative to)

typically reference = position reference, but in this case, the sidenav drawer is meant to be positioned relative to the main menu